internal/namespaces/mnq/v1beta1/custom_nats.go

```diff
@@ -0,0 +1,134 @@
+package mnq
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/scaleway/scaleway-cli/v2/internal/core"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type natsContext struct {
+	Description string `json:"description"`
+	URL         string `json:"url"`
+
+	// CredentialsPath is a path to file containing credentials
+	CredentialsPath string `json:"creds"`
+}
+
+func natsContextFrom(account *mnq.NatsAccount, credsPath string) []byte {
+	ctx := &natsContext{
+		Description:     "Nats context created by Scaleway CLI",
+		URL:             account.Endpoint,
+		CredentialsPath: credsPath,
+	}
+	b, _ := json.Marshal(ctx)
+
+	return b
+}
+
+type CreateContextRequest struct {
+	NatsAccountID   string
+	ContextName     string
+	CredentialsName string
+	Region          scw.Region
+}
+
+func aliasCreateContextCommand() *core.Command {
+	return &core.Command{
+		Short:     "Create a new context for natscli",
+		Namespace: "mnq",
+		Resource:  "nats",
+		Verb:      "create-context",
+		Groups:    []string{"workflow"},
+		Long: `This command help you configure your nats cli
+Contexts should are stored in $HOME/.config/nats/context
+Credentials and context file are saved in your nats context folder with 0600 permissions`,
+		Examples: []*core.Example{
+			{
+				Short: "Create a custom alias 'isl' for 'instance server list'", // TODO
+				Raw:   `scw alias create isl command="instance server list""`,
+			},
+			{
+				Short: "Add an alias to a verb",
+				Raw:   `scw alias create c command=create`,
+			},
+		},
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:       "nats-account-id", // TODO should be optional and an interactive list should show accounts
+				Required:   true,
+				Positional: true,
+				Short:      "Alias name",
+			},
+			{
+				Name:  "name",
+				Short: "Name of the saved context, defaults to account name",
+			},
+			{
+				Name:    "credentials-name",
+				Short:   "Name of the created credentials",
+				Default: core.RandomValueGenerator("mnq"),
+			},
+			core.RegionArgSpec((*mnq.NatsAPI)(nil).Regions()...),
+		},
+		ArgsType: reflect.TypeOf(CreateContextRequest{}),
+		Run: func(ctx context.Context, argsI interface{}) (interface{}, error) {
+			args := argsI.(*CreateContextRequest)
+			api := mnq.NewNatsAPI(core.ExtractClient(ctx))
+
+			natsAccount, err := api.GetNatsAccount(&mnq.NatsAPIGetNatsAccountRequest{
+				Region:        args.Region,
+				NatsAccountID: args.NatsAccountID,
+			}, scw.WithContext(ctx))
+			if err != nil {
+				return nil, fmt.Errorf("failed to get nats account: %w", err)
+			}
+
+			creds, err := api.CreateNatsCredentials(&mnq.NatsAPICreateNatsCredentialsRequest{
+				Region:        args.Region,
```

Comment:
We mandatorily create credentials for him. If he already has some, the user could provide us with a credentials ID, and we could retrieve them instead of creating new ones each time. What do you think?